component/instantiation/HNNInstantiated.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component, createRef } from 'react';
+import Button from '@material-ui/core/Button';
+import { withStyles } from '@material-ui/core';
+
+import Utils from '../../Utils';
+import ErrorDialog from './ErrorDialog';
+import Plots from '../general/materialComponents/Plots';
+import Canvas from '../../../../js/components/interface/3dCanvas/Canvas';
+import IconButton from '../../../../js/components/controls/iconButton/IconButton';
+import MaterialIconButton from '../general/materialComponents/IconButtonWithTooltip';
+import ControlPanel from '../../../../js/components/interface/controlPanel/controlpanel';
+
+const styles = {
+  instantiatedContainer: {
+    height: '100%',
+    width: '100%',
+    left: "0px",
+    top: "32px",
+    position: 'fixed',
+  },
+  controlpanelBtn: {
+    position: 'absolute',
+    left: 34,
+    top: 320
+  },
+  plotBtn: {
+    position: 'absolute',
+    left: 34,
+    top: 374
+  },
+  refreshButton: {
+    position: 'absolute',
+    right: "10px",
+    top: "45px",
+  },
+  launchButton: {
+    position: 'absolute',
+    right: "10px",
+    top: "90px"
+  },
+  goBackBtn: {
+    position: 'absolute',
+    left: 34,
+    top: 500
+  },
+
+};
+
+class HNNInstantiated extends Component {
+  state = {
+    modelExist: false,
+    canvasUpdateRequired: false,
+    simulationUpdateRequired: true,
+    errorMessage: '',
+    errorDetails: '',
+    openErrorDialog: false
+  };
+  canvasRef = createRef();
+
+
+  componentDidMount() {
+    this.canvasRef.current.engine.setLinesThreshold(10000);
+    this.canvasRef.current.displayAllInstances();
+  }
+
+  async refreshCanvas() {
+    const { simulationUpdateRequired } = this.state;
+    if (simulationUpdateRequired) {
+      await this.instantiate()
+    }
+    GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, GEPPETTO.Resources.PARSING_MODEL);
```

Comment:
@tarelli the use of GEPPETTO globals along the code is something that I'd get rid of, at least inside components

Reply:
@filippometacell that will probably be possible once we refactor how manage events in a post backbone world, what do you think?

Reply:
@tarelli for sure this is not something we can begin with this PR. Maybe a step by step approach could be adopted from some point on. Given that React tend to become fastly a mess when components do business logic on their own a good starting point could be to avoid event handling at all and delegate to a controller class that handles a gross grained, data-centered, functionality. This controller could at first work with GEPPETTO constants and backbone events as it's always been. From that, it should be straighforward to use a consistent backbone-based approach (which is not necessarely bad, I think) or to Redux actions.

Reply:
That would be very sensible

Reply:
I'm closing this one waiting for the upraise of post backbone warriors